### PR TITLE
Hot water zone changes

### PR DESCRIPTION
**Pull request by TeresaRoberts**
This is an updated version of @storeyleon's PR.

This code adds the Hot water accessory, which shows the current state of the hot water switch (on/off) within tado and allows HK to override the schedule to turn the hot water on or off.  The termination will use the default termination set in the plugin or a user configured termination using zone 0.

The accessory shows as a simple on/off switch.  There are no further variables available.  This switch is mains powered as part of the tado extension kit, so there's no battery state.

### CODE REVIEW

File: src/tado-hotwater-zone.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+/**
+  * Can be called to update the zone. Not needed for hot water.
+  */
+TadoHotWaterZone.prototype.updateZone = function (apiZones) { }
+
+/**
+ * Defines the export of the file.
+ */
+module.exports = TadoHotWaterZone;
```
Comment (lukasroegner) [inline src/tado-hotwater-zone.js]: Newline at end-of-file recommended

**Comment by lukasroegner** [comment]
That looks nice, thank you! Do you have the hardware from Tado at home to test out the implementation?

**Comment by Cardo1** [comment]
> That looks nice, thank you! Do you have the hardware from Tado at home to test out the implementation?

Yes, we do.  We’re currently using this branch and it works as expected.

This is how it works:

If tado schedule is hot water off, then the switch shows as off in HK.  If the user switches it on in HK, it will turn on the hot water for the predefined length (as per the global default or specific zone termination option) by applying an overlay.  Turning it off again in HK turns the hot water off (back to the tado schedule, deleting the overlay).

If tado schedule is hot water on, then it shows as on in HK.  However, turning hot water off in HK doesn’t actually do anything, as all it does is deletes the overlay, leaving the hot water on as per the schedule (the switch flips back to on when it subsequently reads the state).

We’ve considered the different ways of dealing with this, however can’t think of a better way to handle it.  If the hot water is off, the user might want to turn it on to heat up the water (for a bath or shower, for example), however if the hot water is already on, it’s unlikely the user would want to turn it off, manually.  HK doesn’t have an elegant way to show an On/Off/Auto switch where it shows the On/Off state whilst in Auto.

File: src/tado-hotwater-zone.js (hunk 1 above)
Comment (TeresaRoberts) [reply src/tado-hotwater-zone.js]: I've added the newline.

**Comment by lukasroegner** [comment]
New version on NPM!

**Comment by omarassil** [comment]
I'm having an issue with this feature described in the link here. It's not working to turn the hot water ON
https://github.com/lukasroegner/homebridge-tado-smart-thermostat/issues/16

**Pull request merged**